### ISSUE TITLE
Does not work on MNIST

### ISSUE THREAD

**Issue by ghost** [gabrielrcouto/php-convnet#1]
Does not work on MNIST.

There are two examples in the archive.
One loads a working Javascript version of the neural network, the other creates its own in PHP.
In getPrediction, it always returns the same value on a trained neural network.

During training, a bug with softmax:
in SoftmaxLayer::backward, in return -log($this->es[$y]); -- $y instance of ConvNet\Vol instead single value.

[demo](https://github.com/gabrielrcouto/php-convnet/files/1240578/phpconvnet.zip)

**Comment by fremsoft** [comment]
I've found a bug on Vol class:

on line #46  change this:
  $this->w[$i] = rand(0.0, $scale);
into this:
  $this->w[$i] = **(rand()*$scale)/getrandmax();**

This allows weights to be scaled random numbers instead of Zeroes

Then another critical Bug on SigmoidLayer and ReluLayer, on forward function let use reference for $V2w:

...
$V2w = **&**$V2->w;
...
